feat: Setup storybook and add ProposalCard stories.

mprasanjith opened the pull request:
# Description

Adds storybook and implements the configurations needed to get Guilds components working.

Also adds an example story for the `ProposalCard` component.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested locally.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any UI changes have been tested and made responsive for mobile views

mprasanjith committed:
feat: Setup storybook and add ProposalCard stories.

rossneilson commented:
This is great, just need the fixtures stuff we talked about in standup. Also moving stories and fixtures files up into the individual component folders

MiltonTulli reviewed (dismissed):
Looks good to me!
Another cool thing we can add is preview deployment of storybook generated page on each pr. Not necessary in this pr.

mprasanjith committed:
feat: Setup Storybook GH Actions deployment.

mprasanjith committed:
feat: Setup Storybook GH Actions deployment.

mprasanjith committed:
fix: Build failing due to missing localhost config.

mprasanjith committed:
fix: Cleanup Github deploy script.

mprasanjith committed:
feat: Switch to Cloudflare IPFS gateway for deploy previews.

mprasanjith committed:
fix: Issue in storybook deploy.

mprasanjith committed:
feat: Cleanup storybook deployment scripts.

mprasanjith committed:
chore: Restructure stories and fixtures.

mprasanjith commented:
I've moved the stories and fixtures as we discussed. Also added the Github pipeline for Storybook. It won't deploy until the storybook-deploy.yml file is in the main branch though. For now, you can see it working on my fork [here](https://github.com/mprasanjith/dxvote/pull/36).

rossneilson commented on src/Components/ProposalCard/types.ts:
```diff
@@ -6,5 +6,5 @@ export interface ProposalCardProps {
   ensAvatar?: ENSAvatar;
   href?: string;
-  statusProps: ProposalStatusProps;
+  statusProps?: ProposalStatusProps;
```
Need to remember to make render this as a child and not prop drill
